diff --git a/solhint.js b/solhint.js
index 755a910c9c084d027520a73b7b989b9b7912a975..53d912394404ee6bc0a4f283da9ee145f1f73d76 100755
--- a/solhint.js
+++ b/solhint.js
@@ -160,8 +160,7 @@ function executeMainActionLogic() {
 
   printReports(reports, formatterFn)
 
-  // exitWithCode(reports)
-  process.exit(0)
+  exitWithCode(reports)
 }
 
 function processStdin(options) {
@@ -182,8 +181,7 @@ function processStdin(options) {
   const reports = [report]
 
   printReports(reports, formatterFn)
-  process.exit(0)
-  // exitWithCode(reports)
+  exitWithCode(reports)
 }
 
 function writeSampleConfigFile() {
@@ -363,10 +361,10 @@ function listRules() {
   }
 }
 
-// function exitWithCode(reports) {
-//   const errorsCount = reports.reduce((acc, i) => acc + i.errorCount, 0)
-//   process.exit(errorsCount > 0 ? 1 : 0)
-// }
+function exitWithCode(reports) {
+  const errorsCount = reports.reduce((acc, i) => acc + i.errorCount, 0)
+  process.exit(errorsCount > 0 ? 1 : 0)
+}
 
 function checkForUpdate() {
   // eslint-disable-next-line import/no-extraneous-dependencies
